apps/ejabberd/src/http_upload/mod_http_upload.erl

```diff
@@ -103,14 +192,30 @@ with_default_backend(Opts) ->
     end.
 
 
--spec module_opts() -> proplists:proplist().
-module_opts() ->
-    gen_mod:get_module_opts(?MYNAME, ?MODULE).
+-spec module_opts(ejabberd:server()) -> proplists:proplist().
+module_opts(Host) ->
+    gen_mod:get_module_opts(Host, ?MODULE).
+
+
+-spec generate_token(ejabberd:server()) -> binary().
+generate_token(Host) ->
+    base16:encode(crypto:strong_rand_bytes(token_bytes(Host))).
 
 
--spec generate_token() -> binary().
-generate_token() ->
-    base16:encode(crypto:strong_rand_bytes(token_bytes())).
+-spec file_too_large_error(MaxFileSize :: non_neg_integer()) -> jlib:exml().
+file_too_large_error(MaxFileSize) ->
+    MaxFileSizeBin = integer_to_binary(MaxFileSize),
+    Error0 = ?ERR_NOT_ACCEPTABLE,
+    Error0#xmlel{
```

Comment:
This whole record update is not very readable to me. Probably assigning the additional child to a var and updating the record in one line will make things more readable here. At least for me :)